compute/wps/backends/cdat.py

```diff
@@ -215,18 +220,22 @@ def configure_preprocess(self, variable, domain, operation, user, job, **kwargs)
 
                 analysis[uri].insert(0, map_domain)
 
-            canvas = canvas | celery.group(celery.chain(x) for x in analysis.values())
+        canvas = canvas | celery.group(celery.chain(x) for x in analysis.values())
+
+        job.steps_inc_total(len(analysis) * len(analysis.values()[0]))
 
         # Setup the task to submit the job for actual execution.
         execute = tasks.wps_execute.s(
             variable, domain, operation, user_id=user.id, job_id=job.id).set(
                 **helpers.DEFAULT_QUEUE)
 
+        job.steps_inc_total()
+
         canvas = canvas | execute
 
         canvas.delay()
 
-    def generate_ingress_tasks(self, op_uid, url, var_name, user_id, job_id, **kwargs):
+    def generate_ingress_tasks(self, op_uid, url, var_name, user_id, job_id, kwargs):
```

Comment:
I'm curious why you remove the ** here.

Reply:
I needed to pass the dict in rather than spread it. I'm storing a state variable in it.